### PR TITLE
[4.0] Convert icon- in toolbar to fa- ( part 1 )

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue #27333 

### Summary of Changes
change icon- classes to be consistent with using fontawesome 5 classes. ( part 1 )

### Testing Instructions
apply pr
run npm ci
edit an article. you will see all the icons in the toolbar.
verify that icons have changed properly and none are missing.

### Expected result
Icons are same as before the pr.

### Documentation Changes Required

none

### CODE REVIEW

**Comment by brianteeman** [comment]
are you planning to convert all the instances of icon- ??

seems a waste of effort to do this as the icon that is rendered is the fontawesome icon

**Comment by Quy** [comment]
Yes, please read the following comment: https://github.com/joomla/joomla-cms/pull/27333#issuecomment-568258299

**Comment by N6REJ** [comment]
> are you planning to convert all the instances of icon- ??
> 
> seems a waste of effort to do this as the icon that is rendered is the fontawesome icon

because of things like this I'm not sure it's entirely realistic to try to change all in a single pr..  "icon-new" is a great example of the problem.
image

**Comment by N6REJ** [comment]
@Quy I think we need to wait for #27657 to get merged before moving forward so we are not chasing our tail

**Comment by C-Lodder** [comment]
> seems a waste of effort to do this as the icon that is rendered is the fontawesome icon

@brianteeman The `icon-*` mappings were there for extension developers to maintain B/C. Ideally they should be made deprecated in 4.0 and removed in 5.0. So moving core to the FA classes would be best done now rather than later.

Of course, the deprecation for `icon-*` is just my opinion. If it were up to me, those mappings wouldn't even be there.

**Comment by N6REJ** [comment]
> > seems a waste of effort to do this as the icon that is rendered is the fontawesome icon
> 
> @brianteeman The `icon-*` mappings were there for extension developers to maintain B/C. Ideally they should be made deprecated in 4.0 and removed in 5.0. So moving core to the FA classes would be best done now rather than later.
> 
> Of course, the deprecation for `icon-*` is just my opinion. If it were up to me, those mappings wouldn't even be there.

I checked with @wilsonge and can confirm they will be gone in J! 5

**Comment by brianteeman** [comment]
so just mark them as deprecated and get on with something more useful. it really is just renaming for the sake of renaming

**Comment by N6REJ** [comment]
> so just mark them as deprecated and get on with something more useful. it really is just renaming for the sake of renaming

so, your suggestion is to mark them deprecated but NOT fix them?
Is that truly constructive?.

**Comment by brianteeman** [comment]
They're not brokwn

**Comment by N6REJ** [comment]
right, but only because we've bastardized fontawesome & when we do remove them in 5 all this work will need to be done.

**Comment by brianteeman** [comment]
Where have we done that?

**Comment by N6REJ** [comment]
image
you'll see here we've added icon-options to fontawesome

**Comment by N6REJ** [comment]
thats just one example

**Comment by brianteeman** [comment]
That is what is generated by the scss - nothing wrong there at all - it is **exactly** what you are supposed to do.

 As @C-Lodder said before it is there for backwards compatibility so that developers do not have to change all their icons if they dont want to.

**Comment by N6REJ** [comment]
no, that is vendor file.  It should be sacrosanct and what icomoon map is for.  Then that is compiled into template.css

File: administrator/components/com_postinstall/tmpl/messages/default.php [hunk 1]
```diff
@@ -50,7 +50,7 @@
 		<h2><?php echo Text::_('COM_POSTINSTALL_LBL_NOMESSAGES_TITLE'); ?></h2>
 		<p><?php echo Text::_('COM_POSTINSTALL_LBL_NOMESSAGES_DESC'); ?></p>
 		<a href="<?php echo Route::_('index.php?option=com_postinstall&view=messages&task=message.reset&eid=' . $this->eid . '&' . $this->token . '=1'); ?>" class="btn btn-warning btn-lg">
-			<span class="icon icon-eye-open" aria-hidden="true"></span>
+			<span class="icon fas fa-eye" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_postinstall/tmpl/messages/default.php]: Remove `icon`?

**Comment by brianteeman** [comment]
It is not a vendor file.

**Comment by C-Lodder** [comment]
@N6REG that's not the vendor. The mappings extend the same styling using SCSS's `@extend` method. You'll notice a comma between the classes by which they both use the styling from that file ;)

Either way, using FA classes in core now isn't hurting anyone. No reason not to do it now, then in a couple of decades when J5 is ready, the SCSS file simply needs removing.

**Comment by brianteeman** [comment]
Thanks @c-lodder. As @n6rej doesn't believe anything I say hopefully he will listen to you. I just don't have the patience to explain how scss works to a front-end developer

**Comment by N6REJ** [comment]
@C-Lodder this is why I'm saying its a vendor file..
image
I see the icomoon.scss which is doing mapping... 
image
so, are you saying, we're taking icomoon.scss, fontawesome 5 scss & putting the compiled version in vendor/fontawesome.css ?
why in the vendor folder?  Thats confusing.
btw, I saw the comma so I understand how that part is working.

File: administrator/components/com_postinstall/tmpl/messages/default.php (hunk 1 above)
Comment (N6REJ) [reply administrator/components/com_postinstall/tmpl/messages/default.php]: not sure.  I've not researched how that class is being used there yet.  It's not used everywhere icon- is used to I felt it safer to leave it, at least for now.

**Comment by N6REJ** [comment]
> Thanks @C-Lodder. As @N6REJ doesn't believe anything I say hopefully he will listen to you. I just don't have the patience to explain how scss works to a front-end developer

@brianteeman because MOST of the time instead of being constructive & helpful,  you are negative and condescending towards ANYTHING I try to do.

**Comment by C-Lodder** [comment]
@N6REJ  have a look at this file (which belongs to the template):  https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/vendor/fontawesome-free/fontawesome.scss

You'll notice it imports the FA vendor and icomoon mapping. It is then compiled as a separate file to the template CSS: https://github.com/joomla/joomla-cms/blob/4.0-dev/build/build-modules-js/compilecss.es6.js#L55

FA do not use any `icon-` prefixes

**Comment by N6REJ** [comment]
> FA do not use any `icon-` prefixes
100% correct.

**Comment by N6REJ** [comment]
This is part 1 of the "fix".  Reason for stopping here is because the next step is going to be much more complicated to fix as its not simple changes.  So I figure better to have separate pr for that one.

**Comment by N6REJ** [comment]
how do I reset this to draft?

File: administrator/components/com_redirect/layouts/toolbar/batch.php [hunk 2]
```diff
@@ -17,6 +17,6 @@
 
 ?>
 <button type="button" data-toggle="modal" onclick="{document.getElementById('collapseModal').open(); return true;}" class="btn btn-sm btn-primary">
-	<span class="icon-checkbox-partial" aria-hidden="true"></span>
+	<span class="<span class="fas fa-square" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_redirect/layouts/toolbar/batch.php]: Fix. You have `span` within `span`.

File: administrator/components/com_modules/tmpl/modules/default.php [hunk 3]
```diff
@@ -122,7 +122,7 @@
 							<?php else : ?>
 								<?php // Extension is not enabled, show a message that indicates this. ?>
 								<span class="tbody-icon" title="<?php echo Text::sprintf('COM_MODULES_MSG_MANAGE_EXTENSION_DISABLED', $this->escape($item->name)); ?>">
-									<span class="icon-ban-circle" aria-hidden="true"></span>
+									<<span class="fas fa-minus-circle" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_modules/tmpl/modules/default.php]: Remove extra `<` in `<<span`

File: administrator/components/com_redirect/layouts/toolbar/batch.php (hunk 2 above)
Comment (N6REJ) [reply administrator/components/com_redirect/layouts/toolbar/batch.php]: thanks

**Comment by N6REJ** [comment]
READY FOR TESTING
I'm leaving icon-new and icon-option and maybe a couple more for another pr since its not simple <span class=
its icon-$name and it's not clear to me where & how that is done YET.

File: modules/mod_login/tmpl/default.php [hunk 4]
```diff
@@ -79,15 +79,15 @@
 						<input id="modlgn-secretkey-<?php echo $module->id; ?>" autocomplete="off" type="text" name="secretkey" class="form-control" placeholder="<?php echo Text::_('JGLOBAL_SECRETKEY'); ?>">
 						<span class="input-group-append" title="<?php echo Text::_('JGLOBAL_SECRETKEY_HELP'); ?>">
 							<span class="input-group-text">
-								<span class="icon-help" aria-hidden="true"></span>
+								<<span class="fas fa-question" aria-hidden="true"></span>
```
Comment (Quy) [inline modules/mod_login/tmpl/default.php]: Remove extra `<`

File: modules/mod_login/tmpl/default.php [hunk 5]
```diff
@@ -79,15 +79,15 @@
...
 							</span>
 						</span>
 					</div>
 				<?php else : ?>
 					<label for="modlgn-secretkey-<?php echo $module->id; ?>"><?php echo Text::_('JGLOBAL_SECRETKEY'); ?></label>
 					<input id="modlgn-secretkey-<?php echo $module->id; ?>" autocomplete="off" type="text" name="secretkey" class="form-control" placeholder="<?php echo Text::_('JGLOBAL_SECRETKEY'); ?>">
 					<span class="btn width-auto" title="<?php echo Text::_('JGLOBAL_SECRETKEY_HELP'); ?>">
-						<span class="icon-help" aria-hidden="true"></span>
+						<<span class="fas fa-question" aria-hidden="true"></span>
```
Comment (Quy) [inline modules/mod_login/tmpl/default.php]: Remove extra `<`

File: modules/mod_login/tmpl/default.php (hunk 4 above)
Comment (N6REJ) [reply modules/mod_login/tmpl/default.php]: ty

File: administrator/components/com_joomlaupdate/tmpl/joomlaupdate/default_preupdatecheck.php [hunk 6]
```diff
@@ -48,7 +48,7 @@
 								<span class="badge badge-<?php echo $option->state ? 'success' : 'danger'; ?>">
 									<?php echo Text::_($option->state ? 'JYES' : 'JNO'); ?>
 									<?php if ($option->notice) : ?>
-										<span class="icon-info icon-white" title="<?php echo $option->notice; ?>"></span>
+										<span class="fas fa-info"icon-white" title="<?php echo $option->notice; ?>"></span>
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/tmpl/joomlaupdate/default_preupdatecheck.php]:
```suggestion
										<span class="fas fa-info icon-white" title="<?php echo $option->notice; ?>"></span>
```

File: components/com_content/tmpl/archive/default_items.php [hunk 7]
```diff
@@ -124,7 +124,7 @@
 					<?php if ($params->get('show_hits')) : ?>
 						<dd>
 							<div class="hits">
-								<span class="icon-eye-open"></span>
+								<span class="fas fa-eye"></span>
```
Comment (Quy) [inline components/com_content/tmpl/archive/default_items.php]:
```suggestion
								<span class="fas fa-eye" aria-hidden="true"></span>
```

File: components/com_content/tmpl/archive/default_items.php [hunk 8]
```diff
@@ -214,7 +214,7 @@
 				<?php if ($params->get('show_hits')) : ?>
 					<dd>
 						<div class="hits">
-							<span class="icon-eye-open"></span>
+							<span class="fas fa-eye"></span>
```
Comment (Quy) [inline components/com_content/tmpl/archive/default_items.php]:
```suggestion
							<span class="fas fa-eye" aria-hidden="true"></span>
```

File: components/com_users/tmpl/login/default_logout.php [hunk 9]
```diff
@@ -43,7 +43,7 @@
 		<div class="com-users-logout__submit control-group">
 			<div class="controls">
 				<button type="submit" class="btn btn-primary">
-					<span class="icon-arrow-left icon-white"></span>
+					<span class="fas fa-arrow-alt-circle-left icon-white"></span>
```
Comment (Quy) [inline components/com_users/tmpl/login/default_logout.php]:
```suggestion
					<span class="fas fa-arrow-alt-circle-left icon-white" aria-hidden="true"></span>
```

File: components/com_users/tmpl/profile/default.php [hunk 10]
```diff
@@ -27,7 +27,7 @@
 		<ul class="com-users-profile__edit btn-toolbar float-right">
 			<li class="btn-group">
 				<a class="btn" href="<?php echo Route::_('index.php?option=com_users&task=profile.edit&user_id=' . (int) $this->data->id); ?>">
-					<span class="icon-user"></span> <?php echo Text::_('COM_USERS_EDIT_PROFILE'); ?>
+					<span class="fas fa-user"></span> <?php echo Text::_('COM_USERS_EDIT_PROFILE'); ?>
```
Comment (Quy) [inline components/com_users/tmpl/profile/default.php]:
```suggestion
					<span class="fas fa-user" aria-hidden="true"></span> <?php echo Text::_('COM_USERS_EDIT_PROFILE'); ?>
```

File: installation/tmpl/preinstall/default.php [hunk 11]
```diff
@@ -73,7 +73,7 @@
 							</div>
 							<div class="form-group row">
 								<div class="col-md-8 offset-md-2 justify-content-end d-flex">
-									<button id="verifybutton" class="btn btn-success"><span class="icon-ok icon-white"></span> <?php echo Text::_('INSTL_VERIFY_FTP_SETTINGS'); ?></button>
+									<button id="verifybutton" class="btn btn-success"><span class="fas fa-check icon-white"></span> <?php echo Text::_('INSTL_VERIFY_FTP_SETTINGS'); ?></button>
```
Comment (Quy) [inline installation/tmpl/preinstall/default.php]:
```suggestion
									<button id="verifybutton" class="btn btn-success"><span class="fas fa-check icon-white" aria-hidden="true"></span> <?php echo Text::_('INSTL_VERIFY_FTP_SETTINGS'); ?></button>
```

File: libraries/cms/html/grid.php [hunk 12]
```diff
@@ -257,7 +257,7 @@ public static function order($rows, $image = 'filesave.png', $task = 'saveorder'
 	{
 		return '<a href="javascript:saveorder('
 			. (count($rows) - 1) . ', \'' . $task . '\')" rel="tooltip" class="saveorder btn btn-sm btn-secondary float-right" title="'
-			. Text::_('JLIB_HTML_SAVE_ORDER') . '"><span class="icon-menu-2"></span></a>';
+			. Text::_('JLIB_HTML_SAVE_ORDER') . '"><span class="fas fa-sort"></span></a>';
```
Comment (Quy) [inline libraries/cms/html/grid.php]:
```suggestion
			. Text::_('JLIB_HTML_SAVE_ORDER') . '"><span class="fas fa-sort" aria-hidden="true"></span></a>';
```

File: modules/mod_login/tmpl/default.php [hunk 13]
```diff
@@ -114,7 +114,7 @@
 			<?php if ($usersConfig->get('allowUserRegistration')) : ?>
 				<li>
 					<a href="<?php echo Route::_($registerLink); ?>">
-					<?php echo Text::_('MOD_LOGIN_REGISTER'); ?> <span class="icon-arrow-right"></span></a>
+					<?php echo Text::_('MOD_LOGIN_REGISTER'); ?> <span class="fas fa-arrow-alt-circle-right"></span></a>
```
Comment (Quy) [inline modules/mod_login/tmpl/default.php]:
```suggestion
					<?php echo Text::_('MOD_LOGIN_REGISTER'); ?> <span class="fas fa-arrow-alt-circle-right" aria-hidden="true"></span></a>
```

File: plugins/system/stats/layouts/field/uniqueid.php [hunk 14]
```diff
@@ -44,5 +44,5 @@
 ?>
 <input type="hidden" name="<?php echo $name; ?>" id="<?php echo $id; ?>" value="<?php echo htmlspecialchars($value, ENT_QUOTES, 'UTF-8'); ?>">
 <button class="btn btn-secondary" type="button" id="js-pstats-reset-uid">
-	<span class="icon-refresh"></span> <?php echo Text::_('PLG_SYSTEM_STATS_RESET_UNIQUE_ID'); ?>
+	<span class="fas fa-sync"></span> <?php echo Text::_('PLG_SYSTEM_STATS_RESET_UNIQUE_ID'); ?>
```
Comment (Quy) [inline plugins/system/stats/layouts/field/uniqueid.php]:
```suggestion
	<span class="fas fa-sync" aria-hidden="true"></span> <?php echo Text::_('PLG_SYSTEM_STATS_RESET_UNIQUE_ID'); ?>
```

**Comment by jwaisner** [comment]
@N6REJ Based on the referenced issue the PR doesnt seem to change the issue with the icons in the toolbar. It doesnt break anything but the gap at the bottom still exits.



Please do correct me if I misunderstand the intended result.

**Comment by Quy** [comment]
@jwaisner That is a separate issue to address later. This PR is to convert `icon` to `fa`. See https://github.com/joomla/joomla-cms/pull/27333#issuecomment-568258299

File: administrator/components/com_associations/Field/Modal/AssociationField.php [hunk 15]
```diff
@@ -67,7 +67,7 @@ protected function getInput()
 			. ' data-select="' . Text::_('COM_ASSOCIATIONS_SELECT_TARGET') . '"'
 			. ' data-change="' . Text::_('COM_ASSOCIATIONS_CHANGE_TARGET') . '"'
 			. ' data-target="#associationSelect' . $this->id . 'Modal">'
-			. '<span class="icon-file" aria-hidden="true"></span> '
+			. '<span class="fas fa-check" aria-hidden="true"></span> '
```
Comment (Quy) [inline administrator/components/com_associations/Field/Modal/AssociationField.php]: This should be a file icon and not a check icon. Fix in other places.

Comment (N6REJ) [reply administrator/components/com_associations/Field/Modal/AssociationField.php]: what in the world...

Comment (N6REJ) [reply administrator/components/com_associations/Field/Modal/AssociationField.php]: fixed https://github.com/joomla/joomla-cms/pull/27686/commits/f5160527d12a4d2e004ff43144c980f834ce6c74

Comment (Quy) [reply administrator/components/com_associations/Field/Modal/AssociationField.php]: There are more like this to be changed.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on b913de361a1ea709e16fd4be317360ed9fb61013<br /><br />Tested all icons throughout including a clean install. No issues found with the display related to this PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27686">issues.joomla.org/tracker/joomla-cms/27686</a>.</sub>

File: administrator/components/com_categories/src/Field/Modal/CategoryField.php [hunk 16]
```diff
@@ -160,7 +160,7 @@ function jSelectCategory_" . $this->id . "(id, title, object) {
 				. ' data-toggle="modal"'
 				. ' type="button"'
 				. ' data-target="#ModalSelect' . $modalId . '">'
-				. '<span class="icon-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
+				. '<span class="fas fa-check" aria-hidden="true"></span> ' . Text::_('JSELECT')
```
Comment (Quy) [inline administrator/components/com_categories/src/Field/Modal/CategoryField.php]:
```suggestion
				. '<span class="fas fa-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
```

File: administrator/components/com_contact/src/Field/Modal/ContactField.php [hunk 17]
```diff
@@ -149,7 +149,7 @@ function jSelectContact_" . $this->id . "(id, title, object) {
 				. ' data-toggle="modal"'
 				. ' type="button"'
 				. ' data-target="#ModalSelect' . $modalId . '">'
-				. '<span class="icon-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
+				. '<span class="fas fa-check" aria-hidden="true"></span> ' . Text::_('JSELECT')
```
Comment (Quy) [inline administrator/components/com_contact/src/Field/Modal/ContactField.php]:
```suggestion
				. '<span class="fas fa-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
```

File: administrator/components/com_content/src/Field/Modal/ArticleField.php [hunk 18]
```diff
@@ -150,7 +150,7 @@ function jSelectArticle_" . $this->id . "(id, title, catid, object, url, languag
 				. ' data-toggle="modal"'
 				. ' type="button"'
 				. ' data-target="#ModalSelect' . $modalId . '">'
-				. '<span class="icon-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
+				. '<span class="fas fa-check" aria-hidden="true"></span> ' . Text::_('JSELECT')
```
Comment (Quy) [inline administrator/components/com_content/src/Field/Modal/ArticleField.php]:
```suggestion
				. '<span class="fas fa-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
```

File: administrator/components/com_newsfeeds/src/Field/Modal/NewsfeedField.php [hunk 19]
```diff
@@ -151,7 +151,7 @@ function jSelectNewsfeed_" . $this->id . "(id, title, object) {
 				. ' data-toggle="modal"'
 				. ' type="button"'
 				. ' data-target="#ModalSelect' . $modalId . '">'
-				. '<span class="icon-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
+				. '<span class="fas fa-check" aria-hidden="true"></span> ' . Text::_('JSELECT')
```
Comment (Quy) [inline administrator/components/com_newsfeeds/src/Field/Modal/NewsfeedField.php]:
```suggestion
				. '<span class="fas fa-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
```

File: administrator/components/com_menus/src/Field/Modal/MenuField.php [hunk 20]
```diff
@@ -282,7 +282,7 @@ function jSelectMenu_" . $this->id . "(id, title, object) {
 				. ' data-toggle="modal"'
 				. ' type="button"'
 				. ' data-target="#ModalSelect' . $modalId . '">'
-				. '<span class="icon-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
+				. '<span class="fas fa-check" aria-hidden="true"></span> ' . Text::_('JSELECT')
```
Comment (Quy) [inline administrator/components/com_menus/src/Field/Modal/MenuField.php]:
```suggestion
				. '<span class="fas fa-file" aria-hidden="true"></span> ' . Text::_('JSELECT')
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b913de361a1ea709e16fd4be317360ed9fb61013<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27686">issues.joomla.org/tracker/joomla-cms/27686</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27686">issues.joomla.org/tracker/joomla-cms/27686</a>.</sub>

**Comment by wilsonge** [comment]
Thanks guys. Nice work @N6REJ

**Comment by N6REJ** [comment]
ty george

**Pull request merged**